packages/wallet/src/redux/protocols/ledger-top-up/reducer.ts

```diff
@@ -195,45 +199,47 @@ const waitForDirectFundingForAReducer: ProtocolReducer<states.LedgerTopUpState>
   sharedData: SharedData,
   action: LedgerTopUpAction,
 ): ProtocolStateWithSharedData<states.LedgerTopUpState> => {
-  if (!isDirectFundingAction(action)) {
-    console.warn(`Received non direct funding action in ${protocolState.type} state.`);
-    return { protocolState, sharedData };
-  }
-
-  const {
-    protocolState: directFundingState,
-    sharedData: directFundingSharedData,
-  } = directFundingStateReducer(protocolState.directFundingState, sharedData, action);
-
-  sharedData = directFundingSharedData;
-
-  const lastCommitment = helpers.getLatestCommitment(protocolState.ledgerId, sharedData);
-  const { ledgerId, processId, proposedAllocation, proposedDestination } = protocolState;
-
-  if (directFundingState.type === 'DirectFunding.FundingFailure') {
-    return { protocolState: states.failure({ reason: 'DirectFundingFailure' }), sharedData };
-  } else if (directFundingState.type === 'DirectFunding.FundingSuccess') {
-    const { consensusUpdateState, sharedData: newSharedData } = initializeConsensusState(
-      TwoPartyPlayerIndex.B,
-      processId,
-      ledgerId,
-      proposedAllocation,
-      proposedDestination,
-      lastCommitment.allocation,
-      protocolState.protocolLocator,
+  if (isDirectFundingAction(action)) {
+    let directFundingState: DirectFundingState;
+    ({ protocolState: directFundingState, sharedData } = directFundingStateReducer(
+      protocolState.directFundingState,
       sharedData,
-    );
+      action,
+    ));
 
-    const newProtocolState = states.restoreOrderAndAddBTopUpUpdate({
-      ...protocolState,
-      consensusUpdateState,
-    });
-    return {
-      protocolState: newProtocolState,
-      sharedData: newSharedData,
-    };
+    const { protocolLocator, processId } = protocolState;
+    if (directFundingState.type === 'DirectFunding.FundingFailure') {
```

Comment:
Maybe `switch` on `directFundingState.type`? I think it's more readable and also more robust, if we use `unreachable`.